include/deal.II/fe/fe_dgq.h

```diff
@@ -312,15 +312,14 @@ class FE_DGQ : public FE_Poly<TensorProductPolynomials<dim>, dim, spacedim>
 
 protected:
   /**
-   * Constructor for tensor product polynomials based on Polynomials::Lagrange
-   * interpolation of the support points in the quadrature rule
-   * <tt>points</tt>. The degree of these polynomials is
-   * <tt>points.size()-1</tt>.
+   * Constructor for tensor product polynomials based on an arbitrary vector
+   * of polynomials. This constructor is used in derived classes to construct
+   * e.g. elements with arbitrary nodes or elements based on Legendre
+   * polynomials.
    *
-   * Note: The FE_DGQ::clone function does not work properly for FE with
-   * arbitrary nodes!
+   * The degree of these polynomials is <tt>polynomials.size()-1</tt>.
    */
-  FE_DGQ (const Quadrature<1> &points);
+  FE_DGQ (const std::vector<Polynomials::Polynomial<double> > &polynomials);
```

Comment:
This constructor is, strictly speaking, part of the public interface of the class because users can access it via derived classes. As such, you may be breaking existing codes here. How hard would it be to retain the old one? I also sort of like the idea of just passing in a quadrature rule and have the class automatically construct the polynomials.